updated time counter and added repo click event

xavluiz opened the pull request:
Added the repo click event

Removed the call clearStatsForPayloadProcess when sending the batch data files. The PluginDataManager processor should be the only one changing the payload stats

Bundled the time counter steps together

Updated the postData to stagger processing the payload if it's an unfocused window

mmclead commented on src/tree/KpmProviderManager.ts:
```diff
@@ -521,6 +516,19 @@ export class KpmProviderManager {
+    getContributorReportButton(identifier: string): KpmItem {
+        const item: KpmItem = new KpmItem();
+        item.label = identifier;
+        item.icon = "icons8-github.svg";
```
the icons have an ENUM defined in the schema now.   Lets check against those values.

xavluiz replied:
wow! :) cool, thanks for adding those

xavluiz commented on src/model/KeystrokeStats.ts:
```diff
@@ -107,6 +108,15 @@ export default class KeystrokeStats {
+            PluginDataManager.getInstance().processPayloadHandler(this, sendNow);
+        } else {
+            // process it in 6 seconds
+            setTimeout(() => {
```
very good point, thank you. It will add 6 seconds to the end times and buffer the elapsed time. should we create the now time here and pass it to the delay call? we can probably get away with 1 delay second for now and keep things as is.

mmclead replied:
for precision we should freeze time here and then send it in.   optional param.   use it if its present otherwise set the time locally.

xavluiz commented:
I've updated the diff to send the "nowTimes" into the processPayloadHandler in case it's the secondary window and updated the icons based on the schema

mmclead commented:
great.   did the prettifier change indentation on those other files?

xavluiz commented:
yeah, the prettier changed formatting. I'd like to see if we can come up with a standard. I'm ok with 2 space indentation and maybe we can use a col width of 100?
